packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts

```diff
@@ -390,15 +422,16 @@ export class S3InputState {
   }
 
   public static getInputPermissionsFromCfnPermissions(selectedGuestPermissions: S3CFNPermissionType[] | undefined) {
-    if (selectedGuestPermissions) {
-      return selectedGuestPermissions.map(S3InputState.getPermissionTypeFromCfnType);
+    if (selectedGuestPermissions && selectedGuestPermissions.length > 0) {
+      const inputParams = selectedGuestPermissions.map(S3InputState.getPermissionTypeFromCfnType);
+      return removeDuplicates(inputParams) as Array<S3PermissionType>; //required to remove List and Read mapping to the same entity
```

Comment:
just convert to Set() and then re-convert to List. I think it's a bit ore readable than the function below.

Reply:
or use loadash